### ISSUE TITLE
Maven: Latest release not available.

### ISSUE THREAD

**Issue by danijoo** [taycaldwell/riot-api-java#61]
It would be awesome to have this library uploaded to jcenter/mavencentral so we can check it out without downloading the jar for every commit.

Best regards,
Daniel

**Comment by alex-place** [comment]
See https://maven.apache.org/guides/mini/guide-central-repository-upload.html

**Comment by briceambrosiak** [comment]
So you mean that it's an enhancement that still to be done, right?
Thanks.

**Comment by goraxan** [comment]
Same problem here

**Comment by recon88** [comment]
Same here...

**Comment by Humberd** [comment]
Same

**Comment by rithms** [comment]
same

**Comment by LogicalOverflow** [comment]
v3.8.2 has sources, but with the season 7 API changes not being in a release yet, it would be nice to have a new release (or snapshots on maven).

**Comment by taycaldwell** [comment]
Availability on maven central is planned when we make a release for 4.0.

In the meantime, feel free to barbarically export a JAR from the source and include it in your project.

**Comment by Gonzo17** [comment]
Hey guys, in Version v3.9.0 are not only the sources missing but also the binaries. So this version can't be used. Version v3.8.2 seems fine though.

**Comment by taycaldwell** [comment]
Thanks for the report @Gonzo17. There is now an open issue regarding this: https://github.com/taycaldwell/riot-api-java/issues/61

**Comment by JumpingRock** [comment]
Piggybacking off of the previous conversation. Right now I am pulling down and using v3.8.2. I realize I could download the latest code here and hit V3, but unsure what is completed or not, or what will change.

Is the plan for the old functions to all be deprecated and/or removed? Or are you modifying them to simply hit the new endpoints. Just wondering if I will have to rewrite my methods or how much refactoring for new methods will entail.

**Comment by Linnun** [comment]
Technically the current master branch should be complete and working, but we need to do more testing before we can release 4.0 of this wrapper. I'm personally already using the current master branch in a test environment for my production applications, and so far it seems to be stable and working correctly for the methods that I'm actively using.

A lot has changed since the last release. This has various reasons, most significantly the conversion to Riot Api v3.

When converting from v3.8 or v3.9 to v4.0, all of the old method calls will need to be adjusted because methods no longer use a `Region` but a `Platform` parameter now due to the Riot Api v3 drifting away from regions and going towards platforms exclusively. Other than that, batch calls are no longer supported by Riot Api v3, and thus have been removed from this wrapper aswell. Methods might also have their parameters or return DTOs adjusted to reflect other changes to the Riot Api endpoints.


Since old Riot Api methods are deprecated and you will need to do the switch soon either way, I would recommend that you use the latest master branch to create your code and switch to the full release of v4.0.0 later. Please let us know if you happen to find any issues.

**Comment by JumpingRock** [comment]
Thanks for the response @Linnun informative and helpful. I downloaded the master and I'm swapping all my methods over to the latest versions. Luckily I'm not too deep so things won't be too bad.

**Comment by taycaldwell** [comment]
v4.0.0 is available via Jitpack.

Gradle, maven, sbt, and leiningen instructions for getting the project into your build are available [here](https://jitpack.io/#taycaldwell/riot-api-java).
